tests/test_pytensorf.py

```diff
@@ -342,6 +342,140 @@ def test_check_parameters_can_be_replaced_by_ninf(self):
         with pytest.raises(ParameterValueError, match="test"):
             fn([-1, 2, 3])
 
+    def test_check_parameters_removed_from_scan(self):
+        def scan_step(x_0):
+            cond = pt.ge(x_0, 1)
+            x = check_parameters(x_0, cond)
+            x_update = collect_default_updates([x])
+            return x, x_update
+
+        xs, _ = scan(
+            fn=scan_step,
+            sequences=[
+                pt.zeros(3),
+            ],
+            name="xs",
+        )
+
+        with pytest.raises(ParameterValueError):
+            pytensor.function([], xs)()
+
+        with pm.Model() as m:
+            pass
+
+        m.check_bounds = False
+        with m:
+            fn = compile_pymc([], xs)
+            assert np.all(fn() == 0)
+
+        m.check_bounds = True
+        with m:
+            fn = compile_pymc([], xs)
+            assert np.all(fn() == -np.inf)
```

Comment:
Instead of messing with models and `compile_pymc`, let's just use `pytensor.function` directly and pass the rewrites we want.

```python
pytensor.function(..., mode=get_mode().including("rewrite_name"))
```

These tests should be in `logprob/test_utils.py`

Reply:
I will change and move tests